apps/rush-lib/src/logic/deploy/DeployArchiver.ts

```diff
@@ -0,0 +1,97 @@
+import * as JSZip from 'jszip';
+import * as path from 'path';
+import { FileSystem, FileSystemStats, AlreadyExistsBehavior } from '@rushstack/node-core-library';
+import { IDeployState } from './DeployManager';
+
+// JSZip is dependant on Blob being declared.
+declare global {
+  // eslint-disable-next-line
+  type Blob = any;
+}
+export class DeployArchiver {
+  public static async createArchive(deployState: IDeployState): Promise<void> {
+    if (deployState.scenarioConfiguration.json.postCopySourceFolder !== undefined) {
+      const sourceFolderPath: string = FileSystem.getRealPath(
+        deployState.scenarioConfiguration.json.postCopySourceFolder
+      );
+      FileSystem.copyFiles({
+        sourcePath: sourceFolderPath,
+        destinationPath: deployState.targetRootFolder,
+        alreadyExistsBehavior: AlreadyExistsBehavior.Error
+      });
+    }
+    if (deployState.createArchiveFileName !== undefined) {
+      if (!deployState.createArchiveFileName.endsWith('.zip')) {
+        console.log(
+          'Create-archive currently only supports creation of zip files. Skipping create-archive step.\n'
+        );
+        return;
+      }
+      console.log('Invoking "JSZip"...\n');
```

Comment:
This message refers to an implementation detail that won't be meaningful to an end user. Something like this might be better:
```
Creating archive...
```

Same for the `Completed "JSZip" successfully.` message